### ISSUE TITLE
ili2db4: No module named 'packaging'

### ISSUE THREAD

**Issue by nicpitsch** [opengisch/QgisModelBaker#360]
**Error**
When starting QGIS it raises this python error:
```
WARNING    Traceback (most recent call last):
              File "C:/OSGeo4W/apps/qgis-dev/./python\qgis\utils.py", line 334, in _startPlugin
              plugins[packageName] = package.classFactory(iface)
              File "C:/Users/user1/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\QgisModelBaker\__init__.py", line 23, in classFactory
              from .qgismodelbaker import QgisModelBakerPlugin
              File "C:/OSGeo4W/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "C:/Users/user1/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\QgisModelBaker\qgismodelbaker.py", line 24, in 
              from QgisModelBaker.gui.generate_project import GenerateProjectDialog
              File "C:/OSGeo4W/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "C:/Users/user1/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\QgisModelBaker\gui\generate_project.py", line 27, in 
              from QgisModelBaker.gui.options import OptionsDialog, CompletionLineEdit
              File "C:/OSGeo4W/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "C:/Users/user1/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\QgisModelBaker\gui\options.py", line 23, in 
              from QgisModelBaker.libqgsprojectgen.db_factory.db_simple_factory import DbSimpleFactory
              File "C:/OSGeo4W/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "C:/Users/user1/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\QgisModelBaker\libqgsprojectgen\db_factory\db_simple_factory.py", line 19, in 
              from .db_factory import DbFactory
              File "C:/OSGeo4W/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "C:/Users/user1/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\QgisModelBaker\libqgsprojectgen\db_factory\db_factory.py", line 21, in 
              from QgisModelBaker.libili2db.ili2dbconfig import Ili2DbCommandConfiguration
              File "C:/OSGeo4W/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "C:/Users/user1/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\QgisModelBaker\libili2db\ili2dbconfig.py", line 21, in 
              from QgisModelBaker.libili2db.ili2dbutils import get_all_modeldir_in_path
              File "C:/OSGeo4W/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
              File "C:/Users/user1/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\QgisModelBaker\libili2db\ili2dbutils.py", line 28, in 
              from packaging import version
              File "C:/OSGeo4W/apps/qgis-dev/./python\qgis\utils.py", line 744, in _import
              mod = _builtin_import(name, globals, locals, fromlist, level)
             ModuleNotFoundError: No module named 'packaging'
```

**Environment**
- QGIS v. 3.4.14 and later.
- QGIS Model Baker 5.0.0.
- OS Windows 10 64 bit.

**Comment by marioba** [comment]
I have the same issue on Windows 7 too. 
@signedav seems related to this change https://github.com/opengisch/QgisModelBaker/blob/7368fb34f2c539fbe5c62cb1ed46db27a6dc8661/QgisModelBaker/libili2db/ili2dbutils.py#L28

**Comment by signedav** [comment]
There has been an issue with a beta release - the upcoming release today ver. 5.0.1 should be fine again. 

But let's keep this issue. It's an issue concerning the current ili2db4 branch.

**Comment by m-kuhn** [comment]
Looks like the [packaging](https://pypi.org/project/packaging/) module is not installed on all systems.

I guess the easiest approach is to ship it together with model baker.

1. Download the packaging wheel from pypi
2. Unzip it into the source tree (maybe a new subdir `external` or `libs`?)
3. Adjust the imports

**Comment by gacarrillor** [comment]
Could we just borrow a function [like this]() instead of shipping the whole packaging library? As far as I know, the only task we do using `packaging` is [this comparison](https://github.com/opengisch/QgisModelBaker/blob/7368fb34f2c539fbe5c62cb1ed46db27a6dc8661/QgisModelBaker/libili2db/ili2dbutils.py#L56).

**Comment by m-kuhn** [comment]
We could write/ship/maintain our own. With the risk that the maintenance overhead is bigger than expected compared to using a widely tested and supported existing module.
What's the downside of shipping packaging?

**Comment by gacarrillor** [comment]
Just edited the link in the previous comment, I had missed it.

>>  What's the downside of shipping packaging?

The size of a complete module vs. a function of 40 lines (could be even reduced for ModelBaker).

**Comment by m-kuhn** [comment]
It's 4k dependencies (even less zipped). We can eventually drop these once we have a dependency management for QGIS windows in place. Compare this to downloading ili2pg.
And it will save us the trouble of dealing with `-beta1` or `-dev1234` suffixes in case they need to be dealt with ...

**Comment by m-kuhn** [comment]
Fixed in [QgisModelBaker-5.0.0-beta2.zip](https://github.com/opengisch/QgisModelBaker/files/4044834/QgisModelBaker-5.0.0-beta2.zip)
